File: src/main/java/net/Chidoziealways/everythingjapanese/event/ModRegistryEvents.java
package net.Chidoziealways.everythingjapanese.event;

import net.Chidoziealways.everythingjapanese.EverythingJapanese;
import net.Chidoziealways.everythingjapanese.jutsu.Jutsu;
import net.Chidoziealways.everythingjapanese.util.ModRegistries;
import net.minecraft.core.HolderLookup;
import net.minecraft.core.Registry;
import net.minecraftforge.eventbus.api.SubscribeEvent;
import net.minecraftforge.fml.common.Mod;
import net.minecraftforge.registries.NewRegistryEvent;
import net.minecraftforge.registries.RegistryBuilder;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.slf4j.Marker;
import org.slf4j.MarkerFactory;

@Mod.EventBusSubscriber(modid = EverythingJapanese.MOD_ID, bus = Mod.EventBusSubscriber.Bus.MOD)
public class ModRegistryEvents {
    private static final Logger log = LoggerFactory.getLogger(ModRegistryEvents.class);
    private static final Marker marker = MarkerFactory.getMarker("MODREGISTRYEVENTS");

    private static final int MAX_VARINT = Integer.MAX_VALUE - 1;

    @SubscribeEvent
    public static void onNewRegistries(NewRegistryEvent event) {
        log.info(marker, "Registering Custom Registries");
        event.create(
                new RegistryBuilder<Jutsu>()
                        .setName(ModRegistries.JUTSU.location())
                        .setDefaultKey(ModRegistries.JUTSU.location())
                        .setMaxID(MAX_VARINT)
        );
        log.info(marker, "Finished Registering Jutsu");
    }

}

File: src/main/kotlin/net/Chidoziealways/everythingjapanese/event/ModRegistryEvents.kt
package net.Chidoziealways.everythingjapanese.event

import net.Chidoziealways.everythingjapanese.JAPANESE_MOD_ID
import net.Chidoziealways.everythingjapanese.block.custom.hanging_scroll.Design
import net.Chidoziealways.everythingjapanese.kanji.KanjiType
import net.Chidoziealways.everythingjapanese.util.ModRegistries
import net.neoforged.bus.api.SubscribeEvent
import net.neoforged.neoforge.registries.DataPackRegistryEvent
import net.neoforged.neoforge.registries.NewRegistryEvent
import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.slf4j.Marker
import org.slf4j.MarkerFactory
import thedarkcolour.kotlinforforge.common.KotlinMod

@KotlinMod.KotlinEventBusSubscriber(modId = JAPANESE_MOD_ID)
object ModRegistryEvents {

    private val log: Logger = LoggerFactory.getLogger(ModRegistryEvents::class.java)
    private val marker: Marker? = MarkerFactory.getMarker("MODREGISTRYEVENTS")

    private const val MAX_VARINT = Int.Companion.MAX_VALUE - 1

    @SubscribeEvent
    fun onNewRegistries(event: NewRegistryEvent) {
        log.info(marker, "Registering Custom Registries")
        event.register(ModRegistries.JUTSU)
        event.register(ModRegistries.MORPHS)

        log.info(marker, "Finished Registering Registries")
    }

    @SubscribeEvent
    fun onNewDatapackRegistries(event: DataPackRegistryEvent.NewRegistry) {
        log.info(marker, "Registering Custom Datapack Registries")

        event.dataPackRegistry(ModRegistries.KANJI, KanjiType.DIRECT_CODEC, KanjiType.DIRECT_CODEC) {builder -> builder.maxId(MAX_VARINT)}
        event.dataPackRegistry(ModRegistries.DESIGN, Design.CODEC, Design.CODEC) { builder -> builder.maxId(MAX_VARINT)}

        log.info(marker, "Finished Registering Datapack Registries")
    }
}
